File: backend/app/schemas/tracked.py
from datetime import datetime

from pydantic import BaseModel


class TrackAnimeRequest(BaseModel):
    anime_id: int
    anime_slug: str
    anime_title: str
    cover_url: str | None = None
    genres: list[str] = []
    plot: str | None = None
    year: str | None = None
    source_site: str = "animeunity"
    check_interval_minutes: int = 60


class TrackedAnimeResponse(BaseModel):
    id: int
    anime_id: int
    anime_slug: str
    anime_title: str
    cover_url: str | None
    source_site: str
    last_known_episode: int
    enabled: bool
    check_interval_minutes: int
    last_checked_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TrackedAnimeUpdate(BaseModel):
    enabled: bool | None = None
    check_interval_minutes: int | None = None


class TrackedListResponse(BaseModel):
    tracked: list[TrackedAnimeResponse]

File: frontend/src/types/tracked.ts
export interface TrackedAnime {
  id: number;
  anime_id: number;
  anime_slug: string;
  anime_title: string;
  cover_url: string | null;
  source_site: string;
  last_known_episode: number;
  enabled: boolean;
  check_interval_minutes: number;
  last_checked_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface TrackAnimeRequest {
  anime_id: number;
  anime_slug: string;
  anime_title: string;
  cover_url?: string | null;
  genres?: string[];
  plot?: string | null;
  year?: string | null;
  source_site?: string;
  check_interval_minutes?: number;
}

export interface TrackedAnimeUpdate {
  enabled?: boolean;
  check_interval_minutes?: number;
}

export interface TrackedListResponse {
  tracked: TrackedAnime[];
}
